File: RAFAEL_ROANI_GONCALVES/src/decorator/without/SMSNotification.java
package decorator.without;

public class SMSNotification extends Notification {

    private String phoneNumber;
    
    public SMSNotification(String message, String phoneNumber) {
        super(message);
        this.phoneNumber = phoneNumber;
    }
    
    @Override
    public void send() {
        super.send();
        System.out.println("Also sending via SMS to: " + phoneNumber);
    }

}

File: ARTHUR TENORIO/FACTORY_METHOD/CERTO/NOTIFIER/SmsNotification.js
const Notification = require("./Notification");

class SmsNotification extends Notification {
  send(recipient, message) {
    if (recipient.length !== 11) {
      console.log("Número inválido para SMS!");
      return;
    }
    console.log(`[SMS] Enviando SMS para ${recipient}`);
    console.log(`Conteúdo: ${message}`);
  }
}

module.exports = SmsNotification;
